src/context/StoreContext.js

```diff
@@ -87,6 +87,20 @@ export default function StoreContextProvider(props) {
     localSetBibleReference(obj)
   }
 
+  const setNewBibleBook = (bId) => {
+    const newBibleBookSourceId = "new-Biblebook"
+    const _bibleRef = bibleReference
+    _bibleRef.sourceId = newBibleBookSourceId
+    if (_bibleRef?.bookId?.toUpperCase() !== bId?.toUpperCase()) {
+      _bibleRef.bookId = bId?.toUpperCase()
+      _bibleRef.chapter = "1"
+      _bibleRef.verse = "1"
+    }
+    bRefActions.applyBooksFilter([bId?.toLowerCase()])
+    bRefActions.goToBookChapterVerse(bId?.toLowerCase(), _bibleRef?.chapter?.toString(), _bibleRef?.verse?.toString())
```

Comment:
is `toString()` necessary here? from line `96` I gather that `chapter`, `verse` are already string primitives (not String objects)

Reply:
Since we are using an RCL here, which I have not taken the time to improve, we have had errors happening, due to chapter and verse not having the right type. So this here is just safeguarding against this - due to that chapter and verse might be coming in through the current bibleReference state.
In other words, until the bible-reference-rcl has ben fixed (the right place to safe guard), then I prefer to safe guard here - based on previous experience...

Reply:
Great could you write that as a comment above this line so that future developers know why that decision was made here?